diff --git a/CHANGELOG.md b/CHANGELOG.md
index 722045091ff49..b6f2ed70c8ba5 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,7 @@ CHANGELOG
 7.4
 ---
 
+ * Add `Command::getCode()` to get the code set via `setCode()`.
  * Allow setting aliases and the hidden flag via the command name passed to the constructor
  * Introduce `Symfony\Component\Console\Application::addCommand()` to simplify using invokable commands when the component is used standalone
  * Deprecate `Symfony\Component\Console\Application::add()` in favor of `Symfony\Component\Console\Application::addCommand()`
diff --git a/Command/Command.php b/Command/Command.php
index 1d2e12bdcce25..1ae758d6a9d20 100644
--- a/Command/Command.php
+++ b/Command/Command.php
@@ -356,6 +356,16 @@ public function complete(CompletionInput $input, CompletionSuggestions $suggesti
         }
     }
 
+    /**
+     * Gets the code that is executed by the command.
+     *
+     * @return ?callable null if the code has not been set with setCode()
+     */
+    public function getCode(): ?callable
+    {
+        return $this->code?->getCode();
+    }
+
     /**
      * Sets the code to execute when running this command.
      *
diff --git a/Command/InvokableCommand.php b/Command/InvokableCommand.php
index 72ff407c81fdf..b497f4df73bfc 100644
--- a/Command/InvokableCommand.php
+++ b/Command/InvokableCommand.php
@@ -30,18 +30,20 @@
  */
 class InvokableCommand implements SignalableCommandInterface
 {
-    private readonly \Closure $code;
+    private readonly \Closure $closure;
     private readonly ?SignalableCommandInterface $signalableCommand;
     private readonly \ReflectionFunction $reflection;
     private bool $triggerDeprecations = false;
+    private $code;
 
     public function __construct(
         private readonly Command $command,
         callable $code,
     ) {
-        $this->code = $this->getClosure($code);
+        $this->code = $code;
+        $this->closure = $this->getClosure($code);
         $this->signalableCommand = $code instanceof SignalableCommandInterface ? $code : null;
-        $this->reflection = new \ReflectionFunction($this->code);
+        $this->reflection = new \ReflectionFunction($this->closure);
     }
 
     /**
@@ -49,7 +51,7 @@ public function __construct(
      */
     public function __invoke(InputInterface $input, OutputInterface $output): int
     {
-        $statusCode = ($this->code)(...$this->getParameters($input, $output));
+        $statusCode = ($this->closure)(...$this->getParameters($input, $output));
 
         if (!\is_int($statusCode)) {
             if ($this->triggerDeprecations) {
@@ -81,6 +83,11 @@ public function configure(InputDefinition $definition): void
         }
     }
 
+    public function getCode(): callable
+    {
+        return $this->code;
+    }
+
     private function getClosure(callable $code): \Closure
     {
         if (!$code instanceof \Closure) {
diff --git a/Tests/Command/CommandTest.php b/Tests/Command/CommandTest.php
index f07b76a360ecc..c8c9166d269e6 100644
--- a/Tests/Command/CommandTest.php
+++ b/Tests/Command/CommandTest.php
@@ -464,6 +464,8 @@ public function testCommandAttribute()
         $this->assertStringContainsString('usage1', $command->getUsages()[0]);
         $this->assertTrue($command->isHidden());
         $this->assertSame(['f'], $command->getAliases());
+        // Standard commands don't have code.
+        $this->assertNull($command->getCode());
     }
 
     #[IgnoreDeprecations]
diff --git a/Tests/Command/InvokableCommandTest.php b/Tests/Command/InvokableCommandTest.php
index 1eac2994485f1..7f11e7674c545 100644
--- a/Tests/Command/InvokableCommandTest.php
+++ b/Tests/Command/InvokableCommandTest.php
@@ -27,6 +27,7 @@
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\NullOutput;
 use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Tests\Fixtures\InvokableTestCommand;
 
 class InvokableCommandTest extends TestCase
 {
@@ -293,6 +294,14 @@ public function __invoke()
         $command->run(new ArrayInput([]), new NullOutput());
     }
 
+    public function testGetCode()
+    {
+        $invokableTestCommand = new InvokableTestCommand();
+        $command = new Command(null, $invokableTestCommand);
+
+        $this->assertSame($invokableTestCommand, $command->getCode());
+    }
+
     #[DataProvider('provideInputArguments')]
     public function testInputArguments(array $parameters, array $expected)
     {
